src/sparseml/yolov8/utils/helpers.py

```diff
@@ -69,3 +77,42 @@ def create_grad_sampler(
         / train_loader.batch_size,
     )
     return grad_sampler
+
+
+def data_from_dataset_path(data: str, dataset_path: str) -> str:
+    """
+    Given a dataset name, fetch the yaml config for the dataset
+    from the Ultralytics dataset repo, overwrite its 'path'
+    attribute (dataset root dir) to point to the `dataset_path`
+    and finally save it to the current working directory.
+    This allows to create load data yaml config files that point
+    to the arbitrary directories on the disk.
+
+     :param data: name of the dataset (e.g. "coco.yaml")
+     :param dataset_path: path to the dataset directory
+     :return: a path to the new yaml config file
+        (saved in the current working directory)
+    """
+    if os.path.basename(dataset_path) != os.path.splitext(data)[0]:
```

Comment:
We don't have to impose this constraint. For instance, I doing something like ~/datasets/my_coco should still be acceptable with coco.yaml.

The dataset path doesn't have to be related to the name of the dataset